Modify rule S2083(Python): Update text to Education Framework structure (APPSEC-200) (#1329)
SonarSource/rspec: rules/S2083/common/fix/function-based-validation.adoc

> ==== Use functions that combine feature and validation

Reviewer: What about "Use secure-by-design API"

Author: ```suggestion
==== Use secure-by-design APIs
```

> The cleanest way to fix this vulnerability is to use a function that combines:

Author: ```suggestion
Some libraries contain APIs these three capabilities:
```

* the retrieval of a file

> * restrict the retrieval of a file to a specific folder

Author: ```suggestion
* restrict the retrieval of a file to a specific folder
* A feature -- such as file download, or file deletion.
```

> Built-in APIs from libraries like {auto_canonicalization_function} benefit from

Author: ```suggestion
Built-in APIs from libraries like `{auto_canonicalization_function}` benefit from
```

> Built-in APIs from libraries like `{auto_canonicalization_function}` benefit from
> quality maintenance regarding features, security, or hardening. It is usually
> better to use these solutions than to develop your own.

Author: ```suggestion
They can be referred to as "secure-by-design" APIs. Using this type of library,
such as '{auto_canonicalization_function}', brings multiple layers of security
to the code while keeping the code base shorter.
```

> The cleanest way to fix this vulnerability is to use a function that combines:
> 
> * the retrieval of a file
> * restrict the retrieval of a file to a specific folder
> 
> Built-in APIs from libraries like `{auto_canonicalization_function}` benefit from
> quality maintenance regarding features, security, or hardening. It is usually
> better to use these solutions than to develop your own.
> 
> Behind the scenes, this function protects against both regular and partial path
> injection.

Reviewer: Isn't it a bit too sujective?
> The cleanest way
> Benefit from quality maintenance

Here is what I would suggest
```suggestion
A common way to fix this vulnerability is to use a secure-by-design API that combines:

* the retrieval of a file
* restrict the retrieval of a file to a specific folder

Built-in APIs like `{auto_canonicalization_function}` were designed to prevent path traversal. It is usually
better to use these solutions than to develop your own.

Behind the scenes, these functions protects against both regular and partial path
injection.
```

Author: Arf, I'm not really comfortable with this suggestion, but I think I'm onto something
